### ISSUE TITLE
Consider weight to determine casting roles

### ISSUE THREAD

**Issue by zedoul** [zedoul/rcalat#1]
rcalat::calc_casting function assumes weights of both source and target tropes are the same --- close to 1. However, that's not true; a character is usually mixture but it does not mean they are equally important. Note that any sufficiently famous character can be everything if there is no weighting.

For example, Lina Inverse may be considered as Melancholic when it comes to some contexts with Gourry Gabriev. However, mostly she is Sanguine.

The goal is this issue is to put weight on tropes, and consider them w.r.t. character to determine similarity. You may think rcalat::compare_char_tropes already considers weight, but that weight is calculated from aggregated values, not from specific character.

In short, what would be better way to weight tropes to calculate casting roles? Think through that;
